compiler.js

```diff
@@ -392,6 +397,17 @@
             return section;
         };
 
+        this.addJS = function (js, story, section, passage, isFirst, inputFilename, lineCount) {
+            if (!passage) {
+                section = this.ensureSectionExists(story, section, isFirst, inputFilename, lineCount);
```

Comment:
Surely though it would be better for the function to follow the same pattern as the others and remain generic rather than being tailored specifically for @deactivate? But given your point regarding inserting JS into the string at all it becomes kind of academic!